modules/xmpp/JingleSessionPC.js

```diff
@@ -549,9 +549,12 @@ export default class JingleSessionPC extends JingleSession {
             const state = this.peerconnection.signalingState;
             const remoteDescription = this.peerconnection.remoteDescription;
 
-            if (browser.usesUnifiedPlan() && state === 'stable'
-                && remoteDescription && typeof remoteDescription.sdp === 'string') {
-                logger.debug(`onnegotiationneeded fired on ${this.peerconnection} in state: ${state}`);
+            logger.info(`${this} onnegotiationneeded fired in state: ${state}`);
+
+            if ((browser.usesUnifiedPlan() || browser.usesPlanBMediaDirectionToStopVideo())
```

Comment:
You are probably right, although I'm not worried about it as I've never seen it fire unnecessarily on plan B, but I will try to do more testing. Eventually I can change the logic to follow the existing scheme in JingleSessionPC.

Reply:
I would rather switch to the existing scheme than checking for a flag what triggered the event.

Reply:
That sounds good! I just checked the webrtc-internals and negotiationneeded is triggered atleast twice in the beginning and in the past there were bug where it would fire it twice for a single track so switching to the existing scheme sounds like a better option.

Reply:
There's only one problem with that. The `setSenderVideoConstraint` is currently called here: https://github.com/jitsi/lib-jitsi-meet/blob/c4c20b9126a0b3fd09384115946495158521b6d4/modules/xmpp/JingleSessionPC.js#L1931
Which is after renegotiation has happened already and it executes inside the JingleSessionPC work queue which means it would get stuck, it won't be possible to wait for any async code to resolve there. Let me know if you have any ideas.

Reply:
I was thinking about switching entire JingleSessionPC to follow the unified model of renegotiating by firing the negotiationneeded event. It would allow as to "debounce" the renegotiations in cases where many things are changing in short time until the state settles down. But it's a big change and I'm afraid of breaking stuff.

Reply:
I agree that would be ideal. Right now, negotiationneeded is getting fired in plan-b whenever a sender is added/removed so those cases will be taken care of by switching to unified model. The only other ones that we would have to handle in JingleSessionPC will be when remote sources are added/removed (which is same in unified plan as well). Yeah, its big change :)

Reply:
I have removed the RN part from the PR for now.